Fixed issue with naming and added support for update

ArturMarschal opened the pull request:
* fixed special charachter issue with reminder and role token naming
* added new features to update
  * added built-in json instead of normal url, as BotC webpage changed
  * added bloodstar-url option, to be able to use tool with homebrew scripts easily
* added small change to name and ability texts, to be able to handle special charcters better, now only ő and ű
* added id for roles, to avoid issue with naming, from now on id is used for token names, not the name, so it is possible to keep id in english but name in localized text

Tsubashi committed:
2025.2.23

Commit:
Fixed issue with naming and added support for update

* fixed special charachter issue with reminder and role token naming
* added new features to update
  * added built-in json instead of normal url, as BotC webpage changed
  * added bloodstar-url option, to be able to use tool with homebrew
    scripts easily
* added small change to name and ability texts, to be able to handle
  special charcters better, now only ő and ű
* added id for roles, to avoid issue with naming, from now on id is used
  for token names, not the name, so it is possible to keep id in english
  but name in localized text

Commit:
Resolved test case to enable wiki images for built-in

Commit:
Updated README

Tsubashi commented on src/botc_tokens/data/known_roles.json:
I definitely appreciate this line of thinking, and I would love to include all the roles by default. Unfortunately, for official roles I'm afraid we aren't allowed to do this. You'll need to remove this file before I could accept the merge.

Source: [the official terms of use](https://bloodontheclocktower.com/legal-terms-of-use#:~:text=You%20do,Institute).

ArturMarschal replied:
I have removed the built in version, as well as the related attribute and code

Tsubashi commented on src/botc_tokens/commands/update.py:
```diff
@@ -33,12 +33,16 @@ def _parse_args():
                         help="Directory in which to write the json and icon files (Default: 'inputs')")
     parser.add_argument('--script-filter', type=str, default='',
                         help="Filter for scripts to pull (Example: 'Experimental')")
+    parser.add_argument('-b', '--bloodstar-url', type=str, default='',
```
It seems to me like it would be better to just upgrade the `--custom-list` argument to be able to tell if the value being passed is a URL and handle it that way. Then you could use it both on bloodstar and on non-bloodstar URLs.

ArturMarschal replied:
I have removed the option and added broader support for custom-list option

Tsubashi commented on src/botc_tokens/commands/update.py:
```diff
@@ -140,16 +145,22 @@ def run():
-    if custom_list:
+    if use_built_in:
+        wiki.role_data = json.load(open(data_dir / "known_roles.json", encoding="utf-8"))
+    elif bloodstar_url.startswith("https://www.bloodstar.xyz"):
```
This is too restrictive. What if someone wanted to use it with one of the unofficial instances (like clocktica.com)? This becomes a non-issue if we just enhance the `--custom-list` argument like I mentioned above.

ArturMarschal replied:
I have opened up to https links

Tsubashi commented on src/botc_tokens/helpers/text_tools.py:
```diff
@@ -173,3 +173,12 @@ def format_filename(in_string):
+        in_string: The string to convert to a font family conform text.
+
+    """
+    return in_string.replace('ő', 'ô').replace('ű', 'û').replace('Ő', 'Ô').replace('Ű', 'Û')
```
I'm not sure we want to be forcing replacements. Since fonts are part of the component bundle, users can swap out the default font if they want to use characters that the default font doesn't support.

ArturMarschal replied:
OK, I have dropped it.

Tsubashi commented:
Would fix #11 and #12. Mentioning here so they get linked.

Tsubashi commented:
Also, could you target the dev branch instead of stable? I use stable to track releases. 

Thanks for your submission!

Commit:
Implemented review comments

Tsubashi commented:
Awesome, I'll review these tomorrow and merge it in if everything looks good.

Tsubashi commented on src/botc_tokens/helpers/wiki_soup.py:
```diff
@@ -31,6 +31,25 @@ def load_from_web(self):
+        if meta_data:
+            version_name = meta_data.get('name')
+        else:
+            print(f"[yellow]Warning:[/] Could not find metadata for the script")
```
Looks like we are missing test coverage on this line. Please add a test that attempts to load a script without metadata.

Tsubashi commented on src/botc_tokens/commands/update.py:
```diff
@@ -150,19 +150,22 @@ def prep_wiki(script_filter, custom_list=None):
-                      f"\n\nI will continue, but [yellow]be warned that it might not work[/].")
-            wiki.role_data = custom_list
+        if custom_list.startswith("https://"):
+            wiki.load_from_web_json(custom_list)
```
I think we should still apply validation whether using local files or URLs. Right now it is being bypassed when using URLs.

Tsubashi reviewed (changes requested):
Don't forget to run the linter as well. Right now there are a few linter issues (line length, import order, etc). 

Thanks for your contribution, and thanks for being cool with my nitpicks. If it gets to be too much just let me know and I can merge it and clean it up myself. 

Again, thanks for making this tool even better!

Commit:
Added review comment fixes

Commit:
Fix import order

ArturMarschal commented:
Sorry, but I did not have time to complete it, but now I think everything has been addressed. Only thing is, that when I added validation to the schema, it already warns, that _meta must be there, so I have removed redundant warning message. Please review. (Btw Im not a python guy, so I'm just trying my best)

Commit:
Fix lint issues
